ui/src/components/RoleForm/index.jsx

```diff
@@ -55,13 +57,19 @@ export default class RoleForm extends Component {
     onRoleDelete: func,
     /** If true, form actions will be disabled. */
     loading: bool,
+    dialogError: oneOfType([string, object]),
+    /**
+     * Callback function fired when the DialogAction component throws an error.
+     * */
+    onDialogActionError: func,
```

Comment:
No need to expose `dialogError` and `onDialogActionError`. I think we can handle the state internally since the dialog is displayed in this component.